addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/model/HttpChannelConfig.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.http.model;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.osgi.framework.BundleContext;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.Optional;
+
+import static org.openhab.binding.http.HttpBindingConstants.DEFAULT_COMMAND_METHOD;
+import static org.openhab.binding.http.HttpBindingConstants.DEFAULT_CONTENT_TYPE;
+import static org.openhab.binding.http.HttpBindingConstants.DEFAULT_STATE_REFRESH_INTERVAL;
+
+/**
+ * A class describing configuration for the HTTP handler.
+ *
+ * @author Brian J. Tarricone
+ */
+@NonNullByDefault
+public class HttpChannelConfig {
+    /**
+     * Enumeration describing the HTTP method.
+     */
+    public enum Method {
+        POST, GET
+    }
+
+    /**
+     * A class describing configuration for the HTTP request to make when fetching {@link State}.
+     *
+     * @author Brian J. Tarricone
```

Comment:
I know that eclipse is auto suggesting this, but inner classes do not have the author tag according to the openHAB coding guidelines.